project_setup.py

```diff
@@ -0,0 +1,253 @@
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""Helps setup a new project based on this template."""
+
+import configparser
+import itertools
+import os
+from typing import Callable, Iterable, Optional
+
+TEMPLATE_PACKAGE_NAME = "package_name"
+TEMPLATE_REPO_URL = "https://github.com/ssec-jhu/base-template"
+MSG_GIT_URL_NOT_FOUND = "--Remote URL not found in .git/config--"
+PROMPT_REMOTE_REPO = "Enter the URL of the remote repository [{url_guess}]: "
+PROMPT_PKG_NAME = "Enter the package name [{name_guess}]: "
+UPDATED_PYPROJECT_TOML = "pyproject.toml [project.urls]"
+UPDATED_README_CICD = "Updated README.md -> CI/Security badges"
+UPDATED_README_RTD = "Updated README.md -> RTD badge"
+UPDATED_README_CODECOV = "Updated README.md -> codecov badge"
+UPDATED_FILE = "Replaced `{template_name}` with `{package_name}` in {file_path} ✅"
+SETUP_COMPLETE = "{package_name} setup complete! 🎉🎉🎉\n"
+
+def make_name_safe(name:str) -> str:
+    """Makes a string safe to use as a package name.
+
+    Args:
+        name (str): The string to make safe.
+
+    Returns:
+        str: The safe string.
+
+    """
+    return name.lower().replace(' ', '_').replace('-', '_')
+
+
+def valid_file_predicate(file_path:str) -> bool:
+    """Checks if a file is valid for modification.
+
+    Args:
+        file_path (str): The path to the file to check.
+
+    Returns:
+        bool: Whether the file is valid for modification.
+
+    """
+    not_in_git = not file_path.startswith("./.git/")
+    not_image =  (
+        not file_path.endswith(".png")
+        and not file_path.endswith(".jpg")
+        and not file_path.endswith(".jpeg")
+    )
+    not_binary = not file_path.endswith(".DS_Store")
+    not_this = not file_path.endswith("project_setup.py")
+
+    return not_in_git and not_image and not_binary and not_this
+
+
+def get_repo_url(
+    file_contents:Iterable[str],
+    input_func:Callable[[Optional[str]], str] = input
+) -> str:
+    """Gets the URL of the remote repository.
+
+    Args:
+        file_contents (Iterable[str]): The contents of the .git/config file.
+        input_func (Callable[[Optional[str]], str], optional): The input function to use. Defaults to input.
+
+    Returns:
+        str: The URL of the remote repository.
+
+    """
+    # cp.read("./.git/config")
+    cp = configparser.ConfigParser()
+    try:
+        cp.read_file(file_contents)
+    except configparser.MissingSectionHeaderError:
+        url_guess = MSG_GIT_URL_NOT_FOUND
+
+    try:
+        url_guess = cp['remote "origin"']["url"]
```

Comment:
perhaps we can get this from git:

```py
subprocess.check_output(['git', 'remote', 'get-url', 'origin']).decode().strip()
```

Reply:
I think using git is a better way to go about it. Then we don't need to be up to date with how git structures its configurations files. I'll change it.

Reply:
I would use [GitPython](https://pypi.org/project/GitPython/) rather than subprocess for this, even if it is adding another dep.

Reply:
Not uninstalled, as that's ephemeral. Could be removed from the deps, however, if added to ``requirements/dev.txt`` or ``requirements/build.txt`` it may not matter. 🤷 
The problem, we'd face is deleting it when it's needed for more than this script, when it's actually wanted - though, perhaps it would be in ``requirements/prd.txt`` then. 🤷